### PR TITLE
[FEAT] Create levelled messages and extract V2 exceptions logic.

### PR DESCRIPTION
**Pull request by roeepolegfiverr**
[FEAT] Create levelled messages and extract V2 exceptions logic.

### CODE REVIEW

**Comment by roeepolegfiverr** [comment]
Init should be:
```
$graylog_notifier = Graylog2Exceptions.new(nil, { hostname: 'graylog-server.service.consul',
                                                  port: '12201',
                                                  facility: 'fiverr-v2-app',
                                                  _environment: Rails.env,
                                                  raise_on_graylog: Settings['raise_on_graylog']})
$graylog_notifier.env_ref = Environment::RequestEnvironment.instance
```

usage:

```
$graylog_notifier.info/warn/error(klass, message, exception)
```

File: lib/graylog2_exceptions.rb [hunk 1]
```diff
@@ -2,9 +2,14 @@
 require 'gelf'
 require 'socket'
 require 'logger'
+require_relative './local_logger'
```
Comment (LitalFiverr) [inline lib/graylog2_exceptions.rb:6]: check if needed

Comment (roeepolegfiverr) [reply lib/graylog2_exceptions.rb:6]: needed.

File: lib/graylog2_exceptions.rb [hunk 2]
```diff
@@ -2,9 +2,14 @@
...
 
 class Graylog2Exceptions
   attr_reader :args
+  attr_writer :env_ref
+
+  FULL_MESSAGE_FIELDS = %w(HTTP_ORIGIN HTTP_REFERER CONTENT_TYPE HTTP_USER_AGENT REMOTE_ADDR REQUEST_URI FIVERR_MESSSAGE)
```
Comment (elyara) [inline lib/graylog2_exceptions.rb]: I think you can add `freeze`

File: lib/graylog2_exceptions.rb [hunk 3]
```diff
@@ -74,19 +79,7 @@ def clean_stack(backtrace)
...
-  end
-
-  def error(err, env=nil)
-    send_to_graylog2(err, env, Logger::ERROR)
-  end
-
-  def send_to_graylog2(err, env=nil, log_level=nil)
+  def send_to_graylog2(err, env = nil, log_level = nil)
     begin
       notifier = GELF::Notifier.new(@args[:hostname], @args[:port], @args[:max_chunk_size])
```
Comment (elyara) [inline lib/graylog2_exceptions.rb]: why won't we reuse this? do we really need to initialize new instance for every message?

Comment (roeepolegfiverr) [reply lib/graylog2_exceptions.rb]: 👍 Old code, but good point, i'll check

File: lib/graylog2_exceptions.rb (hunk 2 above)
Comment (roeepolegfiverr) [reply lib/graylog2_exceptions.rb]: 👍

**Pull request closed without merging**